public/app/plugins/panel/timeseries/migrations.ts

```diff
@@ -345,9 +352,22 @@ export function graphToTimeseriesOptions(angular: any): {
 
   if (angular.stack) {
     graph.stacking = {
-      mode: StackingMode.Normal,
+      mode: angular.percentage ? StackingMode.Percent : StackingMode.Normal,
       group: defaultGraphConfig.stacking!.group,
     };
+
+    if (angular.percentage) {
+      if (angular.yaxis) {
+        delete y1.min;
+        delete y1.max;
+
+        // TimeSeries currently uses 0-1 for percent, so allowing zero leaves only top and bottom ticks.
+        // removing it feels better. probably should fix in TimeSeries, but let's kick it down the road
+        if (y1.decimals === 0) {
+          delete y1.decimals;
+        }
```

Comment:
likely controversial, but this is much simpler than fixing TimeSeries decimals setting in percent stacked mode to understand that 0 decimals is specifically in the 0-100 and not 0-1 range. i think generally this will do the correct thing. let's see what feedback we get before dedicating more time to this than needed.